docker/Dockerfile

```diff
@@ -158,6 +159,16 @@ RUN cd /usr/local && \
     cmake -G Ninja -DCMAKE_INSTALL_PREFIX=${KAHIP_DIR} -DNONATIVEOPTIMIZATIONS=On .. && \
     ninja install
 
+# Install ADIOS2
+ENV ADIOS2_DIR=/usr/local/ADIOS2/deploy
```

Comment:
The steps should be modelled on the PETSc/SLEPc install and cleanup steps rather than Gmsh (Gmsh is a binary install). Also get just the desired tag directly and the git --single-branch option.